src/lib.rs

```diff
@@ -81,7 +81,8 @@ pub struct SubCommand {}
 
 #[allow(deprecated)]
 impl SubCommand {
-    /// Deprecated, replaced with [`App::new`]
+    /// Deprecated, replaced with [`App::new`].
+    /// Did you mean Subcommand (lower-case c)?
     #[deprecated(since = "3.0.0", note = "Replaced with `App::new`")]
```

Comment:
This is only changing the rustdoc and not the compiler deprecation message.  A "did you mean" feels more like its intended for the compiler case.

My suggestions:
```rust
    /// Deprecated, replaced with [`App::new`].  unless you were looking for [Subcommand]
    #[deprecated(since = "3.0.0", note = "Replaced with `App::new` unless you intended the `Subcommand` trait")]
```

Reply:
Thanks, much better.